marlowe-actus/src/Language/Marlowe/ACTUS/Model/SCHED/ContractScheduleModel.hs

```diff
@@ -95,3 +96,52 @@ _SCHED_SC_PAM scfg _IED _SCEF _SCANX _SCCL _MD =
                 | otherwise                           = tt
     in result
 
+-- - Linear Amortizer - --
+_SCHED_IED_LAM = _SCHED_IED_PAM
+
+_SCHED_PR_LAM scfg _PRCL _IED _PRANX _MD =
+    let maybeS  | isNothing _PRANX && isNothing _PRCL = Nothing
+                | isNothing _PRANX                   = Just $ _IED `plusCycle` fromJust _PRCL
+                | otherwise                          = _PRANX
+    in (\s -> _S s (fromJust _PRCL) (fromJust _MD) (scfg { includeEndDay = False })) <$> maybeS
+
+_SCHED_MD_LAM scfg _IED _SD _PRANX _PRNXT _NT _PRCL _MD =
+    let maybeTMinus | isJust _PRANX && ((fromJust _PRANX) >= _SD) = _PRANX
+                    | (_IED `plusCycle` fromJust _PRCL) >= _SD = Just $ _IED `plusCycle` fromJust _PRCL
+                    | otherwise                           = Nothing
+        maybeSupPR | isNothing maybeTMinus = sup (fromMaybe [] $ _SCHED_PR_LAM scfg _PRCL _IED _PRANX _MD) _SD
+                   | otherwise = Nothing
+        result  | isJust _MD = Just [shift scfg $ fromJust _MD]
+                | isJust maybeTMinus = Just [shift scfg $ (fromJust maybeTMinus) `plusCycle` ((fromJust _PRCL) { n = ((ceiling (_NT / (fromJust _PRNXT))) * (n (fromJust _PRCL))) })]
+                | otherwise = Just [(fromJust maybeSupPR) `shiftedPlusCycle` ((fromJust _PRCL) { n = ((ceiling (_NT / (fromJust _PRNXT))) * (n (fromJust _PRCL))) })]
```

Comment:
Here is where the strangeness with the `shiftedPlusCycle` occurs. Only happens when MD and `maybeTMinus` are not defined.